File: backend/src/schemas/preferences.py
"""Schemas for user preference management"""

from __future__ import annotations

from pydantic import BaseModel, Field


class UserPreferenceBase(BaseModel):
    """Shared preference attributes"""

    default_budget_min: int | None = None
    default_budget_max: int | None = None
    preferred_traveler_types: list[str] = Field(default_factory=list)
    preferred_interests: list[str] = Field(default_factory=list)
    avoided_activities: list[str] = Field(default_factory=list)
    dietary_restrictions: list[str] = Field(default_factory=list)
    preferred_accommodation_type: list[str] = Field(default_factory=list)
    mobility_considerations: str | None = None


class UserPreferenceResponse(UserPreferenceBase):
    """API response payload"""

    preferred_pace: str | None = None
    recent_notes: str | None = None
    last_budget_total: int | None = None
    updated_at: str | None = None


class UserPreferenceUpdate(UserPreferenceBase):
    """Update payload for user preferences"""

    pass

File: shared/types/preferences.ts
export interface UserPreferenceResponse {
  default_budget_min: number | null;
  default_budget_max: number | null;
  preferred_traveler_types: string[];
  preferred_interests: string[];
  avoided_activities: string[];
  dietary_restrictions: string[];
  preferred_accommodation_type: string[];
  mobility_considerations: string | null;
  preferred_pace?: string | null;
  recent_notes?: string | null;
  last_budget_total?: number | null;
  updated_at?: string | null;
}

export interface UserPreferenceUpdatePayload {
  default_budget_min: number | null;
  default_budget_max: number | null;
  preferred_traveler_types: string[];
  preferred_interests: string[];
  avoided_activities: string[];
  dietary_restrictions: string[];
  preferred_accommodation_type: string[];
  mobility_considerations: string | null;
}

export const emptyPreferenceResponse: UserPreferenceResponse = {
  default_budget_min: null,
  default_budget_max: null,
  preferred_traveler_types: [],
  preferred_interests: [],
  avoided_activities: [],
  dietary_restrictions: [],
  preferred_accommodation_type: [],
  mobility_considerations: null,
  preferred_pace: null,
  recent_notes: null,
  last_budget_total: null,
  updated_at: null,
}
